src/track/track_gps_l2cm.c

```diff
@@ -0,0 +1,638 @@
+                                    retune loop filters and typically (but
+                                    not necessarily) use longer integration. */
+  alias_detect_t alias_detect; /**< Alias lock detector. */
+  lock_detect_t lock_detect;   /**< Phase-lock detector state. */
+  u16 sat;                     /**< Satellite ID */
+} gps_l2cm_tracker_data_t;
+
+static tracker_t gps_l2cm_trackers[NUM_GPS_L2CM_TRACKERS];
+static gps_l2cm_tracker_data_t gps_l2cm_tracker_data[NUM_GPS_L2CM_TRACKERS];
+
+static void tracker_gps_l2cm_init(const tracker_channel_info_t *channel_info,
+                                  tracker_common_data_t *common_data,
+                                  tracker_data_t *tracker_data);
+static void tracker_gps_l2cm_disable(const tracker_channel_info_t *channel_info,
+                                     tracker_common_data_t *common_data,
+                                     tracker_data_t *tracker_data);
+static void tracker_gps_l2cm_update(const tracker_channel_info_t *channel_info,
+                                    tracker_common_data_t *common_data,
+                                    tracker_data_t *tracker_data);
+
+static bool parse_loop_params(struct setting *s, const char *val);
+static bool parse_lock_detect_params(struct setting *s, const char *val);
+
+static const tracker_interface_t tracker_interface_gps_l2cm = {
+  .code =         CODE_GPS_L2CM,
+  .init =         tracker_gps_l2cm_init,
+  .disable =      tracker_gps_l2cm_disable,
+  .update =       tracker_gps_l2cm_update,
+  .trackers =     gps_l2cm_trackers,
+  .num_trackers = NUM_GPS_L2CM_TRACKERS
+};
+
+static tracker_interface_list_element_t
+  tracker_interface_list_element_gps_l2cm = {
+    .interface = &tracker_interface_gps_l2cm,
+    .next = 0
+  };
+
+/** Register L2 CM tracker into the the tracker interface & settings
+ *  framework.
+ */
+void track_gps_l2cm_register(void)
+{
+  SETTING_NOTIFY(L2CM_TRACK_SETTING_SECTION, "loop_params",
+                 loop_params_string,
+                 TYPE_STRING, parse_loop_params);
+
+  SETTING_NOTIFY(L2CM_TRACK_SETTING_SECTION, "lock_detect_params",
+                 lock_detect_params_string,
+                 TYPE_STRING, parse_lock_detect_params);
+
+  SETTING(L2CM_TRACK_SETTING_SECTION, "cn0_use",
+          track_cn0_use_thres, TYPE_FLOAT);
+
+  SETTING(L2CM_TRACK_SETTING_SECTION, "cn0_drop",
+          track_cn0_drop_thres, TYPE_FLOAT);
+
+  SETTING(L2CM_TRACK_SETTING_SECTION, "alias_detect",
+          use_alias_detection, TYPE_BOOL);
+
+  for (u32 i = 0; i < NUM_GPS_L2CM_TRACKERS; i++) {
+    gps_l2cm_trackers[i].active = false;
+    gps_l2cm_trackers[i].data = &gps_l2cm_tracker_data[i];
+  }
+
+  tracker_interface_register(&tracker_interface_list_element_gps_l2cm);
+}
+
+/** Check if a satellite is already in track
+ *
+ * \param sat Satellite ID
+ * \retval false Satellite is not in track
+ * \retval true Satellite is in track
+ */
+static bool is_in_track(u16 sat)
```

Comment:
Also, now that we're adding tracking startup requests from multiple threads, I realized we need an additional check in `manage_tracking_startup()`:
```
     acq_status_t *acq = &acq_status[sid_to_global_index(startup_params.sid)];
 
+    /* Make sure the SID is not already tracked. */
+    if (acq->state == ACQ_PRN_TRACKING) {
+      continue;
+    }
+
     /* Make sure a tracking channel and a decoder channel are available */
     u8 chan = manage_track_new_acq(startup_params.sid);
```
If you want to copy that into your PR that would be great :)